proofs/tree.go

```diff
@@ -437,29 +437,24 @@ func CalculateProofNodeList(node, leafCount uint64) (nodes []*HashNode, err erro
 	}
 
 	height, _ := merkle.CalculateHeightAndNodeCount(leafCount)
-	index := 0
-	level := height - 1
 	lastNodeInLevel := leafCount - 1
 	offset := uint64(0)
-	nodes = make([]*HashNode, height-1)
+	nodes = make([]*HashNode, 0)
```

Comment:
I actually don't think this is a performance improvement, if the length of the slice is known, it is better to make it with the correct capacity otherwise append() needs to increase the size of the slice. Wouldn't it be better to use height-1 here?

Reply:
@lucasvo I didn't intend the change as a performance improvement. I found having to deal with fewer looping variables helps to understand the code better. But your argument makes sense, since we know the length of the slice is going to be either `height-1` or `height-2`. Also I ran a small benchmark test which shows the performance drop is significant when dealing with larger trees with `append`. I will add the looping index variable again.